### PR TITLE
Chore #157927597 Set Up Tests

### PR DESCRIPTION
**Pull request by ghost**
**What does this PR do?**
Sets Up Testing for TDD

**Description of Task to be completed?**
- Install coveralls and flask-testing
- Add flask-testing code
- Add test coverage using coveralls
- Add Code Climate badge

**How should this be manually tested?**
Clone this repository under the **ch-SetUp-Tests-#157927597 ** branch. You could view the README.md for the badges

**What are the relevant pivotal tracker stories?**
[#157927597](https://www.pivotaltracker.com/story/show/157927597)
[#157931194](https://www.pivotaltracker.com/story/show/157931194)

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,10 +1,16 @@
 aniso8601==3.0.0
 astroid==1.6.4
+certifi==2018.4.16
+chardet==3.0.4
 click==6.7
 colorama==0.3.9
+coverage==4.5.1
+coveralls==1.3.0
+docopt==0.6.2
```
Comment (victoriaaoka) [inline requirements.txt:9]: What are you using this (docopt and colorama) for? considering, you are testing?

File: README.md [hunk 2]
```diff
@@ -17,3 +17,5 @@ This is an application that provides users with the ability to reach out to oper
 
 ### Deployment
 Github pages: https://davenmathews.github.io/Maintenance-Tracker/UI/html/index.html
+
```
Comment (victoriaaoka) [inline README.md:20]: The badges should be at the top of the README. Consider moving them.

File: MaintenanceTrackerAPI/_tests/test_app_users.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        response = self.client.get(api_v1.url_for(UsersSingleRequests, request_id=1))
+        self.assertIn(b'request', response.data)
+
+    def test_a5_User_Creates_Request(self):
+        response = self.client.post(api_v1.url_for(UsersAllRequests))
+        self.assertEqual(201, response.status_code)
+
+    def test_a6_User_Modifies_Request(self):
+        response = self.client.put(api_v1.url_for(UsersSingleRequests, request_id=1))
+        self.assert200(response)
```
Comment (victoriaaoka) [inline MaintenanceTrackerAPI/_tests/test_app_users.py:41]: Good tests but they are all happy tests. Consider handling edge cases in the tests. eg. only logged in users can make/view requests.

**Review by victoriaaoka — COMMENTED** [review]
Good tests but handle edge cases.
Test the requests. 
Test user input eg. when registering and creating/updating requests.
Update the requirements.txt file to have the required requirements only.

File: requirements.txt (hunk 1 above)
Comment (ghost) [reply requirements.txt:9]: I did `pip install pylint` and it came with colorama and other dependencies. I am not sure whether pylint uses them but I for sure don't use them directly. It's like installing Flask and it comes with Werkzeug and urllib3. I don't import them explicitly in my work, but they are needed in Flask.

File: README.md (hunk 2 above)
Comment (ghost) [reply README.md:20]: Will do.

File: MaintenanceTrackerAPI/_tests/test_app_users.py (hunk 3 above)
Comment (ghost) [reply MaintenanceTrackerAPI/_tests/test_app_users.py:41]: These were just set ups. I'm currently writing more tests to handle edge cases as you advised.

File: requirements.txt (hunk 1 above)
Comment (victoriaaoka) [reply requirements.txt:9]: Pylint does not use docopt and colorama. They are actually used to create a CLI interface. 
Are you using a virtual env?

Comment (ghost) [reply requirements.txt:9]: Yes... I am.
